app/Http/Controllers/Api/ChannelController.php

```diff
@@ -0,0 +1,359 @@
+<?php
+
+namespace Suyabay\Http\Controllers\Api;
+
+use DB;
+use Validator;
+use Suyabay\Channel;
+use Suyabay\Episode;
+use Suyabay\Http\Requests;
+use League\Fractal\Manager;
+use Illuminate\Http\Request;
+use League\Fractal\Resource\Item;
+use Illuminate\Mail\Mailer as Mail;
+use League\Fractal\Resource\Collection;
+use Illuminate\Support\Facades\Response;
+use Suyabay\Http\Controllers\Controller;
+use Suyabay\Http\Transformers\ChannelTransformer;
+
+class ChannelController extends Controller
+{
+    protected $mail;
+    protected $response;
+    protected $fractal;
+
+    public function __construct(Manager $fractal, Mail $mail)
+    {
+        $this->fractal = $fractal;
+
+        parent::__construct($mail);
+    }
+
+    /**
+     * This method lists all channels
+     * @param $page
+     * @param $request
+     * @param $response
+     *
+     * @return json $response
+     */
+    public function getAllChannels(Request $request, ChannelTransformer $channelTransformer)
+    {
+        $perPage = $request->query('results') ? : 10;
+
+        $channels = Channel::orderBy('channels.id', 'asc')
+        ->leftJoin('episodes', 'channels.id', '=', 'episodes.channel_id')
+        ->join('users', 'users.id', '=', 'channels.user_id')
+        ->select('channels.*', DB::raw('COUNT(episodes.id) as episodes'))
+        ->groupBy('channels.id')
+        ->skip($this->getRecordsToSkip($perPage, $request))
+        ->take($perPage)
+        ->get();
+
+        $resource = new Collection($channels, $channelTransformer);
+        $data = $this->fractal->createData($resource)->toArray();
+
+        if (count($data['data']) > 0) {
+            return Response::json($data, 200);
+
+        }
+
+        return Response::json(['message' => 'Channels are not available for display'], 404);
+
+    }
+
+    /**
+     * This method get a single channel
+     *
+     * @param $channel_name
+     *
+     * @return json $response
+     */
+    public function getAChannel($channel_name, ChannelTransformer $channelTransformer)
+    {
+        $channel = Channel::where('channel_name', '=', strtolower(urldecode($channel_name)))
```

Comment:
**Code Quality- Nitpick**
@andela-tolotin Looking at what you want to achieve here, this query is looking to complex. Lets discuss about this.